probe-rs/src/probe.rs

```diff
@@ -482,6 +482,14 @@ impl Probe {
     pub fn get_target_voltage(&mut self) -> Result<Option<f32>, DebugProbeError> {
         self.inner.get_target_voltage()
     }
+
+    /// Enable/Disable the Target Power Supply of the probe.
+    ///
+    /// This is not available on all probes.
+    /// This is avialable on some J-Links
+    pub fn try_set_kickstart_power(&mut self, enable: bool) -> Result<(), DebugProbeError> {
```

Comment:
I would love if this didn't include `kickstart` in the name as it's bit weird and J-Link specific :)

Reply:
I'm not sure we want to expose features like this. If this is for a library, we should simplify how to use specific probe drivers. Otherwise, we need a better system to describe probe capabilities and allow configuration through that interface - that would have benefits to all probe drivers, like it would allow implementing SWD over FTDI.

Reply:
My reasoning here is that for example the hs-probe has a feature for this too :)
It's not J-Link specific :)

Reply:
You can also toggle a random FTDI GPIO, I'm not saying the idea is bad, just that it's not general enough for my taste :)

Reply:
we can add functions for probes something like:
`try_into_jlink()-> Jlink{}` 
and provide probe-specific features in the probes

Reply:
That sounds sensible to me :)